src/containers/Book/BookEdit/BookEdit.js

```diff
@@ -0,0 +1,220 @@
+
+    const bookdata = {
+      id: this.props.match.params.id,
+      title,
+      author,
+      status,
+      description,
+      price,
+      discount,
+      key: this.props.record.key,
+    };
+    this.props.updateBookDetails(bookdata);
+    this.props.history.push("/dashboard/books");
+
+  };
+  render() {
+      if (!this.props.record) {
+      return <p>Loading</p>;
+    }
+    const statusOption= [
+      {text: 'Published',value: 'Published'},
+      {text: 'Pending', value: 'Pending'},
+    ]
+    const { title, author, status, description, discount, price } = this.state;
+    return (
+      <div className="sign-margin">
+        <Grid centered>
+          <Grid.Column style={{ maxWidth: 550, marginTop: 20 }}>
+            <Header style={{ color: "#4183c4" }} as="h2">
+              Edit Book
+            </Header>
+
+            <Form size="big" binding={this}>
+              <Form.Input
+                name="title"
+                label="Title"
+                value={title}
+                placeholder="title"
+                onChange={(e) => this.onChange(e)}
+              />
+
+              <Form.Input
+                name="author"
+                label="Author"
+                value={author}
+                placeholder="author"
+                onChange={(e) => this.onChange(e)}
+              />
+
+              {/* <Form.Input
+                name="status"
+                label="Status"
+                value={status}
+                placeholder="status"
+                onChange={(e) => this.onChange(e)}
+              /> */}
+              Status
+               <Dropdown 
+                  label="Status"
+                  placeholder='status'
+                  name="status"
+                  onChange={this.handleChange}
+                  selection 
+                  options={statusOption} 
+                  value={status}
+             />
+
+              <Form.TextArea
+                name="description"
+                label="Description"
+                value={description}
+                type="description"
+                placeholder="description"
+                onChange={(e) => this.onChange(e)}
+              />
+              <Form.Input
+                name="price"
+                label="price"
+                value={price}
+                type="price"
+                placeholder="price"
+                onChange={(e) => this.onChange(e)}
+              />
+              <Form.Input
+                name="discount"
+                label="discount"
+                value={discount}
+                type="discount"
+                placeholder="discount"
+                onChange={(e) => this.onChange(e)}
+              />
+
+              <Button
+                primary
+                //   disabled={isInvalid}
+                control={Button}
+                onClick={this.onSubmit}
+              >
+                Update
+              </Button>
+            </Form>
```

Comment:
Why not formHOC used here. If formHOC created and used in one place, it should be similarly implemented for all the application form.